Sources/Signal.swift

```diff
@@ -1492,6 +1492,83 @@ extension SignalProtocol {
 		}
 	}
 
+	/// Conditionally throttles values sent on the receiver whenever
+	/// `shouldThrottle` is true, forwarding values on the given scheduler.
+	///
+	/// - note: While `shouldThrottle` remains false, values are forwarded on the
+	///         given scheduler.
+	///
+	/// - note: If multiple values are received while `shouldThrottle` is true,
+	///         the latest value is the one that will be passed on.
+	///
+	/// - note: If the input signal terminates while a value is being throttled,
+	///         that value will be discarded and the returned signal will
+	///         terminate immediately.
+	///
+	/// - parameters:
+	///   - shouldThrottle: A boolean property that controls whether values
+	///                     should be throttled.
+	///   - scheduler: A scheduler to deliver events on.
+	///
+	/// - returns: A signal that sends values only while `shouldThrottle` is false.
+	func throttle<P: PropertyProtocol>(while shouldThrottle: P, on scheduler: SchedulerProtocol) -> Signal<Value, Error>
```

Comment:
Hmm, I think I kinda see it the other way. If this was implemented in terms of `Signal`, we'd have to pick a default starting state for the throttle, probably `.resumed` so that events are passed through. Using `Property` means we don't have to make that assumption.

From a slightly different perspective, `Property` is more powerful than `Signal` in that you could implement a signal-based overload of this method by composing it with an initial value and calling through to the property-based implementation.

I pushed 4dcd2bfc92dad2e0f022606a1dc79b86e58d3a96 to demonstrate what I mean. If you think those overloads would be useful I'll add some docs and merge those changes in here.